### PR TITLE
[SW-2461] Add H2O RuleFit to Algo API

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: doc/src/site/sphinx/ml/sw_drf.rst [hunk 1]
```diff
@@ -1,10 +1,17 @@
-.. _drf:
```
Comment (michalkurka) [inline doc/src/site/sphinx/ml/sw_drf.rst:1]: Why do we see changes in sw_drf documentation? Is that intended?

File: doc/src/site/sphinx/ml/sw_rule_fit.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
+.. _drf:
```
Comment (michalkurka) [inline doc/src/site/sphinx/ml/sw_rule_fit.rst:1]: I think the files got mixed up

Comment (mn-mikke) [reply doc/src/site/sphinx/ml/sw_rule_fit.rst:1]: Spot on! Thanks for catching that

**Review by koniecsveta — APPROVED** [review]
LGTM, thanks!

**Pull request merged**